Gems/Atom/Tools/AtomToolsFramework/Code/Source/Util/MaterialPropertyUtil.cpp

```diff
@@ -163,6 +165,88 @@ namespace AtomToolsFramework
         return false;
     }
 
+    bool ConvertToExportFormat(
+        const AZStd::string& exportPath,
+        const AZ::RPI::MaterialTypeSourceData::PropertyDefinition& propertyDefinition,
+        AZ::RPI::MaterialPropertyValue& propertyValue)
+    {
+        if (propertyDefinition.m_dataType == AZ::RPI::MaterialPropertyDataType::Enum && propertyValue.Is<uint32_t>())
+        {
+            const uint32_t index = propertyValue.GetValue<uint32_t>();
+            if (index >= propertyDefinition.m_enumValues.size())
+            {
+                AZ_Error("AtomToolsFramework", false, "Invalid value for material enum property: '%s'.", propertyDefinition.m_name.c_str());
+                return false;
+            }
+
+            propertyValue = propertyDefinition.m_enumValues[index];
+            return true;
+        }
+
+        // Image asset references must be converted from asset IDs to a relative source file path
+        if (propertyDefinition.m_dataType == AZ::RPI::MaterialPropertyDataType::Image)
+        {
+            if (propertyValue.Is<AZ::Data::Asset<AZ::RPI::ImageAsset>>())
+            {
+                const auto& imageAsset = propertyValue.GetValue<AZ::Data::Asset<AZ::RPI::ImageAsset>>();
+                const auto& imagePath = AZ::RPI::AssetUtils::GetSourcePathByAssetId(imageAsset.GetId());
+                propertyValue = GetExteralReferencePath(exportPath, imagePath);
+                return true;
+            }
+
+            if (propertyValue.Is<AZ::Data::Instance<AZ::RPI::Image>>())
+            {
+                const auto& image = propertyValue.GetValue<AZ::Data::Instance<AZ::RPI::Image>>();
+                const auto& imagePath = image ? AZ::RPI::AssetUtils::GetSourcePathByAssetId(image->GetAssetId()) : "";
+                propertyValue = GetExteralReferencePath(exportPath, imagePath);
+                return true;
+            }
+        }
+
+        return true;
+    }
+
+    AZStd::string GetExteralReferencePath(const AZStd::string& exportPath, const AZStd::string& referencePath, const uint32_t maxPathDepth)
+    {
+        if (referencePath.empty())
+        {
+            return {};
+        }
+
+        AZ::IO::BasicPath<AZStd::string> exportFolder(exportPath);
```

Comment:
You know you have used used `AZ::IO::Path` since that is a type alias for `AZ::IO::BasicPath<AZStd::string>` and the raw template type is only used in the Path inplementation